data-protocols/ids/ids-core/src/main/java/org/eclipse/dataspaceconnector/ids/core/util/ConnectorIdUtil.java

```diff
@@ -0,0 +1,60 @@
+/*
+ *  Copyright (c) 2021 - 2022 Fraunhofer Institute for Software and Systems Engineering
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Fraunhofer Institute for Software and Systems Engineering - initial API and implementation
+ *
+ */
+
+package org.eclipse.dataspaceconnector.ids.core.util;
+
+import org.eclipse.dataspaceconnector.ids.spi.types.IdsId;
+import org.eclipse.dataspaceconnector.ids.spi.types.IdsType;
+import org.eclipse.dataspaceconnector.runtime.metamodel.annotation.EdcSetting;
+import org.eclipse.dataspaceconnector.spi.EdcException;
+import org.eclipse.dataspaceconnector.spi.system.ServiceExtensionContext;
+
+/**
+ * Utility class for handling the connector's IDS ID.
+ */
+public class ConnectorIdUtil {
+    
+    @EdcSetting
+    public static final String EDC_IDS_ID = "edc.ids.id";
+    public static final String DEFAULT_EDC_IDS_ID = "urn:connector:edc";
+    
+    private ConnectorIdUtil() {}
+    
+    /**
+     * Returns the connector's ID from its IDS ID. Will use the IDS ID supplied in the configuration,
+     * if it is present, else will use a default IDS ID.
+     *
+     * @param context the context providing the configuration values.
+     * @return the connector's ID.
+     * @throws EdcException if a configuration value is present for the IDS ID but is not a valid URN.
+     */
+    public static String resolveConnectorId(ServiceExtensionContext context) {
+        var value = context.getSetting(EDC_IDS_ID, DEFAULT_EDC_IDS_ID);
+    
+        // Hint: use stringified uri to keep uri path and query
+        var result = IdsId.from(value);
+        if (result.succeeded()) {
+            var idsId = result.getContent();
+            if (idsId.getType() == IdsType.CONNECTOR) {
+                return idsId.getValue();
```

Comment:
I have some doubts about the usage of this connectorId... could this be returned as an `IdsId` and passed around?
Because it seems like in the "sender delegates" we are valuing the `_issuerConnector_` and the `_senderAgent_` of the message with the URI parsing of the value, while in the various responses in the `ResponseUtil` we are putting the whole `urn` parsed at URI. Shouldn't them be always the same (as the latter)?

Reply:
Good catch, had not noticed that before. But you're absolutely right, they should always be the same and in the URN notation. I'll open an issue and address this in a separate PR, as it's probably gonna touch a lot of classes.